chorus_book/src/guide-higher-order-choreography.md

```diff
@@ -23,7 +23,8 @@ When you implement the `Choreography` trait, you have access to the `sub_choreo`
 #     sub_choreo: C,
 # };
 impl<C: Choreography> Choreography for HigherOrderChoreography<C> {
-    fn run(self, op: &impl ChoreoOp) {
+    type L = HNil;
```

Comment:
This section needs rewriting. C's location set must be a subset of the location set of `HigherOrderChoreography` and users must write out the location set in the first line.

Reply:
```suggestion
    type L = hlist!();
```

also, I don't want users to use `HNil` directly. While we cannot prevent users from doing so, I don't want to put this in the documentation.

Reply:
> This section needs rewriting. C's location set must be a subset of the location set of `HigherOrderChoreography` and users must write out the location set in the first line.

Ok, this makes sense. One problem I have is that since `C` is generic, even if we write `type L = C::L`, it doesn't compile and complains that C::L is not a subset of itself.  
```rust
impl<C: Choreography> Choreography for HigherOrderChoreography<C> where {
    type L = C::L;
    fn run(self, op: &impl ChoreoOp<Self::L>) {
        op.call(self.sub_choreo);
    }
}
```

Reply:
@ihaveint Yes, so you must specify the location set of the sub-choreography. Something like this should work:

```rust
impl<C: Choreography<(), L=hlist!(Alice, Bob)>> Choreography for HigherOrderChoreography<C> where {
    type L = hlist!(Alice, Bob);
    fn run(self, op: &impl ChoreoOp<Self::L>) {
        op.call(self.sub_choreo);
    }
}
```